### PR TITLE
copy store through a temporary file

### PR DESCRIPTION
**Pull request by domsj**
Still waiting for CI, but feel free to take a look already.

Tested by manually removing arakoon_0.db from a node and then starting it.
Resulted in the following output:
Jul 19 15:34:58 0928: (main|debug): stat /tmp/arakoon/arakoon_0/head.db
Jul 19 15:34:58 0930: (main|debug): stat /tmp/arakoon/arakoon_0/arakoon_0.db
Jul 19 15:34:58 0931: (main|debug): stat /tmp/arakoon/arakoon_0/arakoon_0.db.tmp
Jul 19 15:34:58 0931: (main|debug): copy_file /tmp/arakoon/arakoon_0/head.db /tmp/arakoon/arakoon_0/arakoon_0.db.tmp
Jul 19 15:34:58 1110: (main|debug): done: copy_file
Jul 19 15:34:58 1111: (main|debug): rename /tmp/arakoon/arakoon_0/arakoon_0.db.tmp -> /tmp/arakoon/arakoon_0/arakoon_0.db

### CODE REVIEW

File: src/node/local_store.ml [hunk 1]
```diff
@@ -79,7 +79,26 @@ let copy_store2 old_location new_location overwrite =
       then
         Logger.info_f_ "Not relocating store from %s to %s, destination exists" old_location new_location
       else
-        File_system.copy_file old_location new_location
+        begin
+          begin
+            if dest_exists
+            then
+              File_system.unlink new_location
```
Comment (domsj) [inline src/node/local_store.ml]: Hmm, this probably isn't needed and is handled by the rename below too right?
(Otoh, I don't think it hurts either?)

Comment (NicolasT) [inline src/node/local_store.ml]: You can't rename to an existing location.

Next to that, unlink might fail if for some reason the provided path is removed already by something else, so might want some error detection.

File: src/node/local_store.ml [hunk 2]
```diff
@@ -79,7 +79,26 @@ let copy_store2 old_location new_location overwrite =
...
+            else
+              Lwt.return ()
+          end >>= fun () ->
+          let tmp_file = new_location ^ ".tmp" in
+          File_system.exists tmp_file >>= fun tmp_exists ->
+          begin
+            if tmp_exists
+            then
+              File_system.unlink tmp_file
```
Comment (NicolasT) [inline src/node/local_store.ml:72]: Same as above

File: src/node/local_store.ml [hunk 3]
```diff
@@ -79,7 +79,26 @@ let copy_store2 old_location new_location overwrite =
...
+            else
+              Lwt.return ()
+          end >>= fun () ->
+          File_system.copy_file old_location tmp_file >>= fun () ->
```
Comment (NicolasT) [inline src/node/local_store.ml:76]: `copy_file` doesn't fsync.

File: src/node/local_store.ml [hunk 4]
```diff
@@ -79,7 +79,26 @@ let copy_store2 old_location new_location overwrite =
...
+          File_system.rename tmp_file new_location
```
Comment (NicolasT) [inline src/node/local_store.ml:77]: Need to `fsync` dirents?

**Comment by domsj** [comment]
I'll reopen this PR against 1.7 after it passed CI

**Pull request closed without merging**